bootloader/lkenv/lkenv.go

```diff
@@ -81,85 +108,187 @@ func copyString(b []byte, s string) {
 	}
 }
 
-func NewEnv(path string) *Env {
-	return &Env{
+func NewEnv(path string, version Version) *Env {
+	e := &Env{
 		path:    path,
 		pathbak: path + "bak",
-		env: SnapBootSelect_v1{
+		version: version,
```

Comment:
my slight preference would be for this to be an actual interface

```
variant envVariant

type envVariant interface {
     setup(*Env)
     get(*Env, key) string
     set(...)
     configureBootPartitions(bootPartLabels []string)
     ....
}
```
with then implementation V1Variant  V2RunVariant etc

Reply:
I implemented this as suggested, hopefully it is more clear now, it actually ended up being rather elegant and is not too much independent code using interface{} in more places. 

the functions do suffer a bit from jumping around from the exported function from lkenv, to the interface implementation in lkenv_{v1,v2}.go back to the common helpers, but I think it's bearable. The size of the diff did not change much however there's a lot more boilerplate which is hopefully easy to review for correctness